### ISSUE TITLE
RTKQ: How to wait for invalidations to complete after a mutation

### ISSUE THREAD

**Issue by KurtGokhan** [reduxjs/redux-toolkit#1779]
When a mutation invalidates a query, it is not always possible to know what that query's next value is going to be. So I want to wait for query to settle before doing something with the new value.

Current workarounds are:
- Wait for the query to update declaratively
- Action listener

What is the recommend way to do this? Is there is a specific API for this?

[Example code](https://codesandbox.io/s/spring-hill-enu02d?file=/src/features/posts/PostDetail.tsx:2575-3007)

I am using the version: 1.9.2

I tried using the lazy query with `unwrap` but calling it right after mutation causes a `ConditionError`. It would be good if lazy query could return the current subscription as promise instead of an error in such cases.

**Comment by markerikson** [comment]
This has come up a couple of other times. I still don't have any particular suggestion here.  Maybe something that involves fetching the list of invalidated tags, and somehow collecting affected entries and waiting for them?

**Comment by KurtGokhan** [comment]
Tried some different ways, but seems like there is no easy way to do this.

I tried using [`getRunningQueryThunk`](https://redux-toolkit.js.org/rtk-query/api/created-api/api-slice-utils#getrunningquerythunk-and-getrunningmutationthunk) and it seemed promising. But for some reason it returns `undefined` even though the query is still running. `getRunningQueriesThunk` also returns an empty array. Maybe they are not supposed to return refetching queries, even though the docs state this:

> This is useful for SSR scenarios to await all queries (or mutations) triggered in any way, including via hook calls or manually dispatching initiate actions.

By the way the issue can be closed, as I guess it is not a common scenario.
